### ISSUE TITLE
Missing per object options and ordering of objects

### ISSUE THREAD

**Issue by wmcshane** [jhonnymertz/java-wkhtmltopdf-wrapper#105]
**Is your feature request related to a problem? Please describe.**
The wkhtmltopdf tool allows:
* Specifying multiple objects and options for each
* The order of these objects, for example: 
    * cover, toc, page; or
    * page, page, toc, page

Example command that uses global options, per object options and specific order of objects:
```
wkhtmltopdf --header-font-size 10 \
cover cover.html \
page foreword.html --header-left "Foreword" \
toc \
page test.html --footer-right "[page] of [topage]" \
output.pdf
```
Format:
```
wkhtmltopdf <GlobalOptions> <[Object] [ObjectOptions]>... <Output>
```

Currently, i'm facing an issue where i have a document that has the following:
* Cover Page
* Amendments Page
* Foreword Page
* Table Of Contents Page
* Document Content Page/s

The current wrapper always puts the table of contents first, removing the ability to specify where in the document it should be placed.
Additionally, the amendments and foreword shouldn't be included in the table of contents, however, the wrapper currently doesn't support per page parameters so specifying `--exclude-from-outline` for these pages only isn't possible.

**Describe the solution you'd like**
Update the following classes to have:
Page:
* Support for various types (cover, page, toc)
* Params object (allow for object specific params)

Pdf:
* List of pages is the order in which objects are added to the final command
* Remove tocParams and use page of type toc (allows specifying location of toc in final document).
* Leave params to be used as global options

The above solution would allow the tool to generate more complicated documents and remove any issues resulting from seperate generations and merging pdfs (Broken bookmarks in this case, details below).

**Describe alternatives you've considered**
I have tried generating out parts of a pdf individually and then merging the resulting pdfs. However, this results in broken bookmark links that no longer link correctly from the table of contents to the content in the final pdf.

I have also tried only using the params list to achieve:
* Custom ordering of objects
* To prevent `getCommandAsArray` limitations of placing the toc first
* Per object options

This does work, however, this is not as intuitive and requires manually writing pages to temp storage, as adding to the page list results in the page being added to the end of the final command and no way to specify commands to follow it.
Additionally, this feels a little hacky like I'm fighting or abusing the wrapper to achieve the desired output, rather than letting the wrapper do the heavy lifting and construction of the final command. 

**Additional context**
I can provide more information or examples if needed. I have also got an implemented solution to the above, I will submit a PR shortly as this may help highlight what i'm trying to achieve.

**Comment by wmcshane** [comment]
Please see related PR https://github.com/jhonnymertz/java-wkhtmltopdf-wrapper/pull/106

**Comment by jhonnymertz** [comment]
Hi @wmcshane thanks for adding such a detailed issue and providing a solution, really appreciate it! I'll push to review and integrate the PR as soon as possible. Thank you so much for your contribution!

**Comment by wmcshane** [comment]
Hi @jhonnymertz, no worries, helps having a starting point! Thank you for the prompt and positive response, I'll attempt to get back to your comments asap!

**Comment by jhonnymertz** [comment]
Solved in #106
